src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+                manufacturerOptional = Optional.ofNullable(manufacturer);
+                manufacturer.setName(name);
+                manufacturer.setCountry(country);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get by index manufacturer from DB in this id: "
+                    + id, e);
+        }
+        return manufacturerOptional;
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String getAllManufacturer = "SELECT * FROM manufacturers WHERE is_deleted = false;";
+        List<Manufacturer> manufacturersList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllFormatsStatement =
+                        connection.prepareStatement(getAllManufacturer)) {
+            ResultSet resultSet = getAllFormatsStatement.executeQuery();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id",Long.class);
+                String name = resultSet.getString("name");
+                String country = resultSet.getString("country");
+                Manufacturer manufacturer = new Manufacturer();
+                manufacturer.setId(id);
+                manufacturer.setCountry(country);
+                manufacturer.setName(name);
+                manufacturersList.add(manufacturer);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
+        }
+        return manufacturersList;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateRequest = "UPDATE manufacturers SET name = ?, country = ?  WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateData =
+                        connection.prepareStatement(updateRequest,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            updateData.setLong(3, manufacturer.getId());
+            updateData.setString(1, manufacturer.getName());
+            updateData.setString(2, manufacturer.getCountry());
+            updateData.executeUpdate();
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer"
+                    + manufacturer + " in DB", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteById =
+                        connection.prepareStatement(deleteRequest,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            deleteById.setLong(1, id);
+            return deleteById.executeUpdate() >= 1;
```

Comment:
[чекліст](https://mate-academy.github.io/jv-program-common-mistakes/java-JDBC/jdbc-intro/JDBC-intro_checklist.html)
```suggestion
            return deleteById.executeUpdate() > 0;
```